qpmodel/stmt.cs

```diff
@@ -626,6 +626,11 @@ LogicNode FilterPushDown(LogicNode plan, bool pushJoinFilter)
                     // it there - shall we try hard to stop query early? Nope, it is no deserved
                     // to poke around for this corner case.
                     //
+                    // One of the expression normalization rules replaces terms like
+                    // COL > NULL, COL < NULL COL <> NULL with a FALSE term and
+                    // if all of the terms where reduced to WHERE FALSE
+                    // the above policy and assert are no longer valid.
```

Comment:
"assert are no longer valid" which assertion - do you mean " Debug.Assert(trueOrFalse)"? If that's the case, you don't mention that in your new code comment.

L649: so trueOrFalse can be false, what if it is false?

Reply:
I actually meant to change the original comment to "if it is a constant true filer, remove it. If a false filter, push it down if possible to help the optimizer eliminate the query node (scan etc) if possible. But left the original comment and added an explanation for your review before finalizing the decision to push a false filter or not.